flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/v2/PaimonWriter.java

```diff
@@ -85,6 +92,28 @@ public PaimonWriter(
                         new ExecutorThreadFactory(
                                 Thread.currentThread().getName() + "-CdcMultiWrite-Compaction"));
         this.serializer = serializer;
+        this.checkpointId = CheckpointIDCounter.INITIAL_CHECKPOINT_ID - 1;
```

Comment:
Is it possible to use `sinkInitContext.getRestoredCheckpointId()` to fetch last succesfully executed checkpointId, instead of saving it in the `PaimonWriterState`?